x/group/keeper/keeper.go

```diff
@@ -278,6 +278,32 @@ func (k Keeper) pruneProposal(ctx sdk.Context, proposalID uint64) error {
 	return nil
 }
 
+// updateProposalStatus iterates through all proposals by group policy index and updates proposal status
+func (k Keeper) updateProposalStatus(ctx sdk.Context) error {
+	proposalIt, err := k.proposalByGroupPolicyIndex.PrefixScan(ctx.KVStore(k.key), nil, nil)
```

Comment:
Here we're updating all proposals. We should only update the proposals of a group policy index that has changes

```
proposalIt, err := k.proposalByGroupPolicyIndex.Get(ctx, groupPolicyAddr.Bytes())
```

where groupPolicyAddr is passed as an argument